pages/1.11/deploying-services/marathon-on-marathon/advanced/index.md

```diff
@@ -336,41 +295,46 @@ In this step, permissions are assigned to the Marathon-on-Marathon instance. Per
+dcos security org users grant ${SERVICE_ACCOUNT_ID} dcos:mesos:master:framework:role:${MESOS_ROLE} create --description "Controls the ability of ${MESOS_ROLE} to register as a framework with the Mesos master"
+dcos security org users grant ${SERVICE_ACCOUNT_ID} dcos:mesos:master:reservation:role:${MESOS_ROLE} create --description "Controls the ability of ${MESOS_ROLE} to reserve resources"
+dcos security org users grant ${SERVICE_ACCOUNT_ID} dcos:mesos:master:volume:role:${MESOS_ROLE} create --description "Controls the ability of ${MESOS_ROLE} to access volumes"
+dcos security org users grant ${SERVICE_ACCOUNT_ID} dcos:mesos:master:reservation:principal:${SERVICE_ACCOUNT_ID} delete --description "Controls the ability of ${SERVICE_ACCOUNT_ID} to reserve resources"
+dcos security org users grant ${SERVICE_ACCOUNT_ID} dcos:mesos:master:task:app_id:/ create --description "Controls the ability to launch tasks"
+dcos security org users grant ${SERVICE_ACCOUNT_ID} dcos:mesos:master:volume:principal:${SERVICE_ACCOUNT_ID} delete --description "Controls the ability of ${SERVICE_ACCOUNT_ID} to access volumes"
 ```
 
-# Step 7 - Install a Non-Native Marathon Instance with Assigned Role
+
+# Step 6: Install a Non-Native Marathon Instance with Assigned Role {.tabs}
 In this step, a non-native Marathon instance is installed on DC/OS with the Mesos role assigned.
 
-1.  Create a custom JSON config file and save as `config.json`. This file is used to install the custom non-native Marathon instance. The JSON file contents vary according to your [security mode](/1.11/security/ent/#security-modes). Replace these variables in the examples with your specific information:
+1.  Create a custom JSON config that will be used to install the custom non-native Marathon instance. The JSON file contents vary according to your [security mode](/1.11/security/ent/#security-modes).
+
+    Make sure you replace all the `${VARIABLES}` in the JSON file with the correct values for your test.
 
-    | Variable | Description |
-    |--------------------------|--------------------------------------------|
-    | `<non-native-marathon>` | Non-native Marathon framework name |
-    | `<service-account-id>` | Non-native Marathon service account |
-    | `<secret-name>` | Secret  |
-    | `<myrole>` | Mesos role |
-    | `<your-repo>` | Private Docker registry repo |
-    | `<your-tag>` | Docker tag |
-    | `<linux-user>` | Linux user  `core` or `centos` |
+    Alternatively, if you have exposed the values for all these variables in your terminal session (as explained in the [Variables Section](#variables-in-example)) you can use the following command to automatically replace all the variables with the correct values. It will save the result file in `marathon-filled.json`:
+
+    ```bash
+    perl -p -e 's/\$\{(\w+)\}/(exists $ENV{$1}?$ENV{$1}:"missing variable $1")/eg' < marathon.json > marathon-filled.json
```

Comment:
I applaud the intent of making the process smoother and making our customers assign those env vars once. But a  perl script? :D I don't know...

Reply:
How about some python (makes it at least more readable):
```
python -c "import os; print(os.path.expandvars(open('marathon.json').read()))" > marathon-filled.json
```